### PR TITLE
CLN: Enable codespell for all files

### LINKED ISSUE #44045: ENH:Possible expansion of codespell to more file types
At the moment, [codespell](https://github.com/pandas-dev/pandas/blob/c30afb23dfa948e3b34b4d71abb16ce455012428/.pre-commit-config.yaml#L15-L20) is enabled only for the following file types: `python, rst, markdown`

As a consequence, typos in various other file types can easily slip through (see #44043 as an example).

Could the expansion of codespell to more file types be beneficial?

### PR DESCRIPTION
**Pull request by luke396**
- [x] closes #44045 (Replace xxxx with the GitHub issue number)

It seems works fine in current main branch.

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
thanks

doesn't look like it
```python
 codespell...............................................................................................Failed
- hook id: codespell
- duration: 1.66s
- exit code: 65

pandas/tests/io/data/xml/cta_rail_lines.kml:14: fo ==> of, for, to, do, go
pandas/tests/io/data/xml/cta_rail_lines.kml:14: DESCRIPTIO ==> DESCRIPTION
pandas/tests/io/data/xml/cta_rail_lines.kml:27: fo ==> of, for, to, do, go
pandas/tests/io/data/xml/cta_rail_lines.kml:27: DESCRIPTIO ==> DESCRIPTION
pandas/tests/io/data/xml/cta_rail_lines.kml:40: fo ==> of, for, to, do, go
pandas/tests/io/data/xml/cta_rail_lines.kml:40: DESCRIPTIO ==> DESCRIPTION
pandas/tests/io/data/xml/cta_rail_lines.kml:53: fo ==> of, for, to, do, go
pandas/tests/io/data/xml/cta_rail_lines.kml:53: DESCRIPTIO ==> DESCRIPTION
pandas/tests/io/data/xml/cta_rail_lines.kml:66: fo ==> of, for, to, do, go
pandas/tests/io/data/xml/cta_rail_lines.kml:66: DESCRIPTIO ==> DESCRIPTION
pandas/_libs/src/ujson/lib/ultrajson.h:190: iteratable ==> iterable
pandas/_libs/src/ujson/lib/ultrajson.h:202: iteratable ==> iterable
pandas/_libs/testing.pyx:190: ambigous ==> ambiguous
pandas/_libs/src/parser/tokenizer.c:1831: housand ==> thousand
pandas/_libs/ops_dispatch.pyx:94: nin ==> inn, min, bin, nine
pandas/_libs/missing.pyx:55: ambigous ==> ambiguous
pandas/_libs/tslibs/strptime.pyx:3: TimeRE ==> timer
pandas/_libs/tslibs/strptime.pyx:11: TimeRE ==> timer
pandas/_libs/tslibs/strptime.pyx:28: TimeRE ==> timer
pandas/_libs/tslibs/strptime.pyx:236: TimeRE ==> timer
pandas/_libs/tslibs/strptime.pyx:536: TimeRE ==> timer
pandas/_libs/tslibs/strptime.pyx:552: TimeRE ==> timer
pandas/_libs/tslibs/strptime.pyx:562: TimeRE ==> timer
pandas/_libs/tslibs/strptime.pyx:571: TimeRE ==> timer
pandas/_libs/hashing.pyx:58: datas ==> data
pandas/_libs/hashing.pyx:100: datas ==> data
web/pandas/config.yml:161: dfine ==> define
WARNING: Decoding file using encoding=utf-8 failed: pandas/tests/io/data/xml/baby_names.xml
WARNING: Trying next encoding iso-8859-1
pandas/_libs/lib.pyx:2653: becase ==> because
pandas/_libs/tslibs/timestamps.pyx:2347: overriden ==> overridden
pandas/_libs/tslibs/timestamps.pyx:2357: overriden ==> overridden
pyproject.toml:276: indendation ==> indentation
pandas/_libs/tslibs/timedeltas.pyx:1109: overrided ==> overrode, overridden
```

how did you run it locally? it should be `pre-commit run codespell --all-files`

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -42,7 +42,13 @@ repos:
...
     hooks:
     -   id: codespell
-        types_or: [python, rst, markdown]
+        exclude: |
+            (?x)
+            pandas/tests/io/data/xml
+            |pandas/_libs/ops_dispatch.pyx
+            |pandas/_libs/tslibs/strptime
+            |web/pandas/config.yml
+
```
Comment (MarcoGorelli) [inline .pre-commit-config.yaml:51]: how about expanding `types_or` instead of excluding certain files? `cython` and `c` should probably be in there

**Comment by luke396** [comment]
I forgot to run it with `--all-files` before.

I make some changes, but some of them may not suitable. I am open to any suggestions or advice.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (luke396) [reply .pre-commit-config.yaml:51]: Thanks for so fast replay!

We should add `cython` and `c`, instead of exclude some files?

Comment (MarcoGorelli) [reply .pre-commit-config.yaml:51]: yup

Comment (luke396) [reply .pre-commit-config.yaml]: This time?

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -42,7 +42,11 @@ repos:
     rev: v2.2.2
     hooks:
     -   id: codespell
-        types_or: [python, rst, markdown]
+        types_or: [python, rst, markdown, cython, c]
+        exclude: |
+            (?x)
+            |pandas/_libs/ops_dispatch.pyx
```
Comment (MarcoGorelli) [inline .pre-commit-config.yaml:48]: why are these excluded?

Comment (luke396) [reply .pre-commit-config.yaml:48]: image

`nin` cause error

Comment (luke396) [reply .pre-commit-config.yaml:48]: should I add some comment to explain them?

Comment (MarcoGorelli) [reply .pre-commit-config.yaml:48]: thanks - probably fine to add both to

https://github.com/pandas-dev/pandas/blob/4ab1c76a4a28aca442410da173f91c42caa7c915/pyproject.toml#L610

then

**Review by MarcoGorelli — APPROVED** [review]
Thanks @luke396 !

**Pull request merged**